core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -146,9 +146,15 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
     case Some(topic) => Map("topic" -> topic)
   }
 
-  val messagesInRate = newMeter(BrokerTopicStats.MessagesInPerSec, "messages", TimeUnit.SECONDS, tags)
-  val bytesInRate = newMeter(BrokerTopicStats.BytesInPerSec, "bytes", TimeUnit.SECONDS, tags)
-  val bytesOutRate = newMeter(BrokerTopicStats.BytesOutPerSec, "bytes", TimeUnit.SECONDS, tags)
+  private val metricTypeMap = new Pool[String, Meter]
```

Comment:
The second type can be `Metric` to be generalized.

Reply:
I tried the `Metric` type but the line
```
topicStats(topic).bytesOutRate.mark(value)
```
caused a build failure because the compiler doesn't recognize `mark` is member of the `Metric` type

Reply:
Ack, let's use `Meter` then.